spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/ModuleAggregationUtils.java

```diff
@@ -0,0 +1,152 @@
+
+import java.util.UUID;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.cloud.stream.annotation.EnableModule;
+import org.springframework.cloud.stream.annotation.Processor;
+import org.springframework.cloud.stream.annotation.Sink;
+import org.springframework.cloud.stream.annotation.Source;
+import org.springframework.cloud.stream.binding.BindableProxyFactory;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.channel.DirectChannel;
+
+/**
+ * @author Marius Bogoevici
+ */
+public class ModuleAggregationUtils {
+
+	private static final Log log = LogFactory.getLog(ModuleAggregationUtils.class);
+
+	public static final String INPUT_CHANNEL_NAME = "input";
+
+	public static final String OUTPUT_CHANNEL_NAME = "output";
+
+	/**
+	 * Supports the aggregation of {@link Source}, {@link Sink} and {@link Processor}
+	 * modules by instantiating and binding them directly
+	 *
+	 * @param parentArgs arguments for the parent (prefixed with '--')
+	 * @param modules a list module classes to be aggregated
+	 * @param moduleArgs arguments for the modules (prefixed with '--")
+	 * @return
+	 */
+	public static ConfigurableApplicationContext runAggregated(String[] parentArgs,
+			Class<?>[] modules, String[][] moduleArgs) {
+		ConfigurableApplicationContext parentContext = createParentContext(parentArgs != null ? parentArgs
+				: new String[0]);
+		runEmbedded(parentContext, modules, moduleArgs);
+		return parentContext;
+	}
+
+	public static ConfigurableApplicationContext runAggregated(Class<?>... modules) {
+		return runAggregated(null, modules, null);
+	}
+
+	/**
+	 * Embeds a group of modules into an existing parent context
+	 *
+	 * @param parentContext the parent context
+	 * @param modules a list of classes, representing root context definitions for modules
+	 * @param args arguments for the modules
+	 */
+	public static void runEmbedded(ConfigurableApplicationContext parentContext,
+			Class<?>[] modules, String[][] args) {
+		SharedChannelRegistry bean = parentContext.getBean(SharedChannelRegistry.class);
+		prepareSharedChannelRegistry(bean, modules);
+		// create child contexts first
+		createChildContexts(parentContext, modules, args);
+	}
+
+	private static ConfigurableApplicationContext createParentContext(String[] args) {
+		SpringApplicationBuilder aggregatorParentConfiguration = new SpringApplicationBuilder();
+		aggregatorParentConfiguration
+				.sources(AggregatorParentConfiguration.class)
+				.web(false)
+				.headless(true)
+				.properties(
+						"spring.jmx.default-domain="
+								+ UUID.randomUUID().toString().replace("-", ""));
```

Comment:
we would need to avoid collisions across "aggregate" apps too (and by the way, for single-module cases we probably should not use the entire coordinate string)